test/e2e/features/pageObjects/tcPage.js

```diff
@@ -0,0 +1,411 @@
+        if (wizardPage1) {
+            for (var i = 1; i < wizardPage1.length; i++) {
+                let caseField = await this.caseCreationApiRes.case_fields.find(caseObj => caseObj.id == wizardPage1[i].case_field_id);
+                if (wizardPage1[i].case_field_id != "Organisation1") {
+                    if (caseField.field_type.type == "Complex") {
+                        if (caseField.field_type.complex_fields.length == 1) {
+                            if (caseField.field_type.complex_fields[0].field_type == "Text") {
+                                fieldIdPresent = $(`#${wizardPage1[i].case_field_id}`);
+                            }
+                        } else {
+                            fieldIdPresent = $(`#${wizardPage1[i].case_field_id}_${wizardPage1[i].case_field_id}`);
+                        }
+                    } else {
+                        fieldIdPresent = $(`#${wizardPage1[i].case_field_id}`);
+                    }
+                    // console.log("Againest api response fields validation::" + await fieldIdPresent);
+                    await BrowserWaits.waitForElement(fieldIdPresent);
+                    expect(await fieldIdPresent.isPresent(), `Case creation ${fieldIdPresent} field should be present`).to.be.true;
+                }
+            };
+        }
+
+    }
+
+    async getCasesApiReq(reqURL) {
+        await Request.withSession(this.userName, this.password);
+        let reqData = { size: 25 }
+        const response = await Request.post(reqURL, reqData, { experimental: true });
+        expect(response.status, `${reqURL} ccd api faild status code: ${response.status}`).to.eql(200);
+        return response.data;
+    }
+
+    async workBasketHeaders(index) {
+        await BrowserWaits.waitForElement(this.searchResultsTopPagination);
+        CucumberReportLogger.AddMessage("starting wait for 2 sec for list to render  : " + new Date().toTimeString());
+        await BrowserWaits.waitForSeconds(2);
+        CucumberReportLogger.AddMessage("wait complete : " + new Date().toTimeString());
+        let thLable = $$("ccd-search-result>table>thead tr th");
+        await BrowserWaits.waitForElement($("ccd-search-result>table>thead tr th"));
+        let count = await thLable.count();
+        // console.log("count:" + count + "index:::" + index);
+        let caseResultsThTitle = [];
+        if (count) {
+            // console.log("inside if count:" + count + "index:::" + index);
+            for (let i = index; i < count; i++) {
+                let thText = thLable.get(i).$$(".search-result-column-label");
+                let text = await thText.getText();
+                caseResultsThTitle.push(`${text}`);
+            }
+            return await caseResultsThTitle;
+        }
+
+    }
+
+    async caseResultsThTitleApiRes(URL) {
+        let response = await this.getCasesApiReq(URL);
+        return await response.columns.map(data => data.label)
+    }
+
+    async getCurrentUrl() {
```

Comment:
is this expected to be getCaseId?

Reply:
No, I'm getting the case results page table header names